lib/commands/reshim.bash

```diff
@@ -90,7 +90,7 @@ write_shim_script() {
 
   if [ -f "$shim_path" ]; then
     if ! grep -x "# asdf-plugin: ${plugin_name} ${version}" "$shim_path" >/dev/null; then
-      sed -i.bak -e "s/exec /# asdf-plugin: ${plugin_name} ${version}\\"$'\n''exec /' "$shim_path"
+      sed -i.bak -e "s|exec |# asdf-plugin: ${plugin_name} ${version}\\"$'\n''exec |' "$shim_path"
```

Comment:
don't allowed filename special character:
- Windows (FAT32, NTFS): Any Unicode except` NUL`, `\`, `/`, `:,` `*`, `?`, `"`,`<`,`>`, `|`. Also, [no space character at the start or end, and no period in the end](https://en.wikipedia.org/wiki/Filename#Comparison_of_filename_limitations).
- Mac(HFS, HFS+): Any valid Unicode except `: `or` /`
- Linux(ext[2-4]): Any byte except `NUL `or `/`

so I think `|` may not be the best character in this scenario, but I can't find the better one~, and most of us won't use `|` as a part of filename

Reply:
What if we used Bash substring replacement for this? The Bash expression can be parameterized. (see https://tldp.org/LDP/abs/html/string-manipulation.html and scroll down to the Substring Replacement header).

Something like this (I haven't had a chance to test it):

```bash
line=$(...)
match='exec '
replacement="# asdf-plugin: ${plugin_name} ${version}\\"$'\n''exec "
updated_line=${line/$match/$repl}
```

Reply:
@Stratus3D @jthegedus 
Hi guys:
  Maybe I have found a better way to solve this problem.
  Insert a new line before the last line rather than regex and replace the previous line with multi-lines.
  the code below:
```
sed -i.bak -e '$i\'$'\n'"# asdf-plugin: ${plugin_name} ${version}" "$shim_path"
```
`$i\'$'\n'` means inserting a new line before the last line. this flag works on both Linux and macOS.  so we insert the plugin info to the new line without regex.
 We could avoid the conflict problem with sed delimiter character by using this approach. 
 could you guys review this change? thx.